autoload/codefmt.vim

```diff
@@ -156,7 +156,67 @@ if !exists('s:gofmt')
   call codefmtlib#AddDefaultFormatter(s:gofmt)
 endif
 
+" Formatter: autopep8
+if !exists('s:autopep8')
+  let s:autopep8 = {
+      \ 'name': 'autopep8',
+      \ 'setup_instructions': 'Install autopep8 and '}
+
+  function s:autopep8.IsAvailable() abort
+    return executable(s:plugin.Flag('autopep8_executable'))
+  endfunction
+
+  function s:autopep8.AppliesToBuffer() abort
+    return &filetype is# 'python'
+  endfunction
+
+
+  ""
+  " Reformat the current buffer with autopep8 or the binary named in
+  " @flag(autopep8_executable), only targeting the range between {startline} and
+  " {endline}.
+  function s:autopep8.FormatRange(startline, endline) abort
+    " Hack range formatting by formatting range individually, ignoring context.
+    let l:cmd = [ s:plugin.Flag('autopep8_executable'), "-"  ]
+    call maktaba#ensure#IsNumber(a:startline)
+    call maktaba#ensure#IsNumber(a:endline)
+    let l:lines = getline(1, line('$'))
+    let l:input = join(l:lines[a:startline - 1 : a:endline - 1], "\n")
```

Comment:
What I was suggesting is instead of using a flag, for now use a script-local variable like
```VimL
let l:executable = s:plugin.Flag('autopep8_executable')
if !exists('s:autopep8_supports_range')
  let l:version_output =
      \ maktaba#syscall#Create([l:executable, '--version']).Call().stderr
  let s:autopep8_supports_range =
      \ matchlist(l:version_output, '\m\Cautopep8 \(\d\+\)\.')[1] >= 1
endif
```
Then, if for some reason a user reports the version check is not right for them and they'd rather configure it manually, it's easy to replace the script-local variable with flag operations.